tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -0,0 +1,205 @@
+assert = chai.assert
+
+class FakeHighlight
+
+  constructor: (@segment, @anchor, @page) ->
+
+  @createFrom: (segment, anchor, page) ->
+    new FakeHighlight segment, anchor, page
+
+# Since the system for registering highlight types have been obliterated,
+# for now, we will do a hard override of the TextHighlight class
+Annotator.TextHighlight = FakeHighlight
+
+class FakeAnchor extends Annotator.Anchor
+
+  _getSegment: -> "html segment for " + @id
+
+  constructor: (manager, annotation, target) ->
+    super manager, annotation, target, 0, 0,
+      "fake quote for" + target.id
+
+    @id = "fake anchor for " + target.id
+
+describe 'Annotator.Plugin.EnhancedAnchoring', ->
+  am = null
+  annotator = null
+  data = null
+  ann1 = null
+  ann2 = null
+  ann3 = null
+  target11 = null
+  target12 = null
+  target2 = null
+  target3 = null
+  anchor11 = null
+  anchor12 = null
+  anchor2 = null
+  anchor3 = null
+  hl11 = null
+  hl12 = null
+  hl2 = null
+  hl3 = null
+
+  initAnchoring = ->
+    annotator =
+      publish: (event) ->
+
+    am = new Annotator.Plugin.EnhancedAnchoring()
+    am.annotator = annotator
+    am.pluginInit()
+
+    am._chooseAccessPolicy()
+
+    am.strategies.push
+      name: "dummy anchoring strategy"
+      code: (annotation, target) ->
+        new FakeAnchor am,annotation, target
+
+    am
+
+  addDummyData = ->
+    target11 =
+      id: "target 1.1"
+    target12 =
+      id: "target 1.2"
+    target2 =
+      id: "target 2"
+    target3 =
+      id: "target 3"
+    ann1 =
+      id: "annotation 1"
+      anchors: []
+    ann2 =
+      id: "annotation 2"
+      anchors: []
+    ann3 =
+      id: "annotation 3"
+      anchors: []
+
+    ann1.target = [target11, target12]
+    ann2.target = [target2]
+    ann3.target = [target3]
+
+    anchor11 = am.createAnchor(ann1, target11).result
+    anchor12 = am.createAnchor(ann1, target12).result
+    anchor2 = am.createAnchor(ann2, target2).result
+    anchor3 = am.createAnchor(ann3, target3).result
+
+    hl11 = anchor11.highlight[0]
+    hl12 = anchor12.highlight[0]
+    hl2 = anchor2.highlight[0]
+    hl3 = anchor3.highlight[0]
+
+  describe "Basics", ->
+
+    beforeEach ->
+      initAnchoring()
+      addDummyData()
+
+    it 'can be initialized', ->
+      assert.ok(am)
+
+    it 'it can do some anchoring', ->
+      assert.equal anchor11.id, "fake anchor for target 1.1"
+
+    it 'Annotations have references to their anchors', ->
```

Comment:
the tests should follow the "it ..." format, so "it adds an anchors property to the annotation" would read better.